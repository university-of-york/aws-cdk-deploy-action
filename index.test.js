/* eslint-disable unicorn/import-index */

const runCdkCommands = require('./src/cdkCommands');

jest.mock('@actions/core', () => ({
    getInput: jest.fn((value) => process.env[value]),
    setFailed: jest.fn(),
}));

jest.mock('./src/cdkCommands');

describe('index', () => {
    afterAll(() => jest.restoreAllMocks());

    it('should call run cdk commands with correct arguments', async () => {
        process.env.AWS_ACCOUNT_ID = '1234';
        process.env.AWS_ROLE_NAME = 'role';
        process.env.AWS_STACK_NAME = 'stack';
        process.env.INFRASTRUCTURE_LOCATION = '.';

        await require('./index.js');

        expect(runCdkCommands).toHaveBeenCalledTimes(1);
        expect(runCdkCommands.mock.calls[0][0]).toMatchSnapshot();
    });
});
