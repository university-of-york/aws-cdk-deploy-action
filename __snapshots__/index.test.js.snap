// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should call run cdk commands with correct arguments 1`] = `
Object {
  "AWS_ACCESS_KEY_ID": "secret_id",
  "AWS_ACCOUNT_ID": "1234",
  "AWS_REGION": "some_region",
  "AWS_ROLE_NAME": "role",
  "AWS_SECRET_ACCESS_KEY": "secret_password",
  "AWS_STACK_NAME": "stack",
  "INFRASTRUCTURE_PATH": ".",
  "SKIP_BOOTSTRAP": "false",
}
`;
